app/lib/link_resolver.rb

```diff
@@ -0,0 +1,47 @@
+class LinkResolver
+  include ActionView::Helpers::UrlHelper
+
+  def initialize(current_register_slug:, register_whitelist:)
+    @current_register_slug = current_register_slug
+    @register_whitelist = register_whitelist
+    @url_helpers = Rails.application.routes.url_helpers
+  end
+
+  def resolve(field, field_value)
+    field_register_slug = field['register']
+    field_name = field['field']
+    field_datatype = field['datatype']
+
+    if field_datatype == 'curie' && field_value.include?(':')
+      resolve_curie(field_value)
+    elsif field_register_slug.present? && field_name != current_register_slug && register_whitelist.include?(field_register_slug)
```

Comment:
very minor but given there are quite a few `&&` conditions here, might be more readable if we split onto multiple lines?